cmd/juju/cloud/add_test.go

```diff
@@ -7,172 +7,256 @@ import (
+			},
+		},
+	}
+	garageMAASCloud = cloudfile.Cloud{
+		Type:      "maas",
+		AuthTypes: []cloudfile.AuthType{"oauth"},
+		Endpoint:  "http://garagemaas",
+	}
 
-	sourceDir := c.MkDir()
-	sourceFile := filepath.Join(sourceDir, "someclouds.yaml")
-	source := `
-clouds:
-  aws:
-    type: ec2
-    auth-types: [ access-key ]
-    regions:
-      us-east-1:
-        endpoint: https://us-east-1.aws.amazon.com/v1.2/
-  localhost:
-    type: lxd
-  homestack:
-    type: openstack
-    auth-types: [userpass, access-key]
-    endpoint: http://homestack
-    regions:
-      london:
-        endpoint: http://london/1.0
-      new-york:
-        endpoint: http://newyork/1.0
-  garage-maas:
-    type: mass
-    auth-types: [oauth]
-    endpoint: http://garagemaas
-`[1:]
-	err = ioutil.WriteFile(sourceFile, []byte(source), 0600)
-	c.Assert(err, jc.ErrorIsNil)
-	return sourceFile
+	manualCloud = cloudfile.Cloud{
+		Type:      "manual",
+		AuthTypes: []cloudfile.AuthType{"manual"},
+		Endpoint:  "192.168.1.6",
+	}
+)
+
+func homestackMetadata() map[string]cloudfile.Cloud {
+	return map[string]cloudfile.Cloud{"homestack": homestackCloud}
 }
 
+func localhostMetadata() map[string]cloudfile.Cloud {
+	return map[string]cloudfile.Cloud{"localhost": localhostCloud}
+}
+
+func awsMetadata() map[string]cloudfile.Cloud {
+	return map[string]cloudfile.Cloud{"aws": awsCloud}
+}
+
+func garageMAASMetadata() map[string]cloudfile.Cloud {
+	return map[string]cloudfile.Cloud{"garage-maas": garageMAASCloud}
+}
+
+// func (s *addSuite) createTestCloudData(c *gc.C) string {
+// 	current := `
+// clouds:
+//   homestack:
+//     type: openstack
+//     auth-types: [userpass, access-key]
+//     endpoint: http://homestack
+//     regions:
+//       london:
+//         endpoint: http://london/1.0
+// `[1:]
+// 	err := ioutil.WriteFile(osenv.JujuXDGDataHomePath("clouds.yaml"), []byte(current), 0600)
+// 	c.Assert(err, jc.ErrorIsNil)
+
+// 	sourceDir := c.MkDir()
+// 	sourceFile := filepath.Join(sourceDir, "someclouds.yaml")
+// 	source := `
+// clouds:
+//   aws:
+//     type: ec2
+//     auth-types: [ access-key ]
+//     regions:
+//       us-east-1:
+//         endpoint: https://us-east-1.aws.amazon.com/v1.2/
+//   localhost:
+//     type: lxd
+//   homestack:
+//     type: openstack
+//     auth-types: [userpass, access-key]
+//     endpoint: http://homestack
+//     regions:
+//       london:
+//         endpoint: http://london/1.0
+//       new-york:
+//         endpoint: http://newyork/1.0
+//   garage-maas:
+//     type: mass
+//     auth-types: [oauth]
+//     endpoint: http://garagemaas
+// `[1:]
+// 	err = ioutil.WriteFile(sourceFile, []byte(source), 0600)
+// 	c.Assert(err, jc.ErrorIsNil)
+// 	return sourceFile
+// }
+
 func (s *addSuite) TestAddBadFilename(c *gc.C) {
-	addCmd := cloud.NewAddCloudCommand()
+	fake := newFakeCloudQueryer()
+	badFileErr := errors.New("")
+	fake.Call("ParseCloudMetadataFile", "somefile.yaml").Returns(map[string]cloudfile.Cloud{}, badFileErr)
```

Comment:
I'm really not a big fan of "fluent" APIs.  They're not used very much in Go, and I think for good reason.  They contain more magic than I really like.  For example, when I first read this, I didn't realize it was actually setting data.. it looks like it's making a call or something.  It would be more clear as just adding a value to a map. IMO.